File: guides/micronaut-patterns-composite/java/src/main/java/example/micronaut/HttpHeaderColorFetcher.java
package example.micronaut;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.http.HttpRequest;
import jakarta.inject.Singleton;

import java.util.Optional;

@Singleton // <1>
public class HttpHeaderColorFetcher implements ColorFetcher {
    @Override
    @NonNull
    public Optional<String> favouriteColor(@NonNull HttpRequest<?> request) {
        return request.getHeaders().get("color", String.class);
    }

    @Override
    public int getOrder() { // <2>
        return 10;
    }
}

File: guides/micronaut-patterns-composite/python/src/example/micronaut/http_header_color_fetcher.py
from jakarta.inject import Singleton
from micronaut.http import HttpRequest

from .color_fetcher import ColorFetcher


@Singleton  # <1>
class HttpHeaderColorFetcher(ColorFetcher):
    def favourite_color(self, request: HttpRequest) -> str | None:
        return request.getHeaders().get("color")

    def getOrder(self) -> int:  # <2>
        return 10
